### DISCUSSION TITLE
WASM building but throwing error fail: Uno.UI.Dispatching.CoreDispatcher[0]

### DISCUSSION THREAD

**Discussion by DclearEE** [unoplatform/uno#10346] [Q&A] [answered]
Hi Uno Community, I have a project that I have built off of the newest SkottieSample.
This project will build for everything but WASM. 
I think the addition of some multithreading code may have broke it for WASM. 

I am using SkiaSharp and there are async threading Tasks that I have running and making API calls/requests.

The Output shows nothing out of the ordinary until it starts to throw "The thread {some Hex} has exited with code 0 (0x0)." forever in a seemingly infinite loop.

This is some of the Output before and after it starts throwing that error.

`Initializing host services...
MONO_WASM: Initializing mono runtime
MONO_WASM: ICU data archive(s) loaded, disabling invariant mode
Done loading the BCL
Loading dependency (./package_9ce66cf1706e80755ee1db68daf77f681959049d/WasmWebSockets)
Loading dependency (./package_9ce66cf1706e80755ee1db68daf77f681959049d/jquery-pep)
Loading dependency (./package_9ce66cf1706e80755ee1db68daf77f681959049d/setImmediate)
Loading dependency (./package_9ce66cf1706e80755ee1db68daf77f681959049d/Uno.UI)
Loading dependency (./package_9ce66cf1706e80755ee1db68daf77f681959049d/SkiaSharp.Views.Uno.Wasm)
Loading dependency (./package_9ce66cf1706e80755ee1db68daf77f681959049d/AppManifest)
Loaded dependency (./package_9ce66cf1706e80755ee1db68daf77f681959049d/AppManifest) - remains 5 other(s).
Loaded dependency (./package_9ce66cf1706e80755ee1db68daf77f681959049d/WasmWebSockets) - remains 4 other(s).
Loaded dependency (./package_9ce66cf1706e80755ee1db68daf77f681959049d/jquery-pep) - remains 3 other(s).
Loaded dependency (./package_9ce66cf1706e80755ee1db68daf77f681959049d/setImmediate) - remains 2 other(s).
Loaded dependency (./package_9ce66cf1706e80755ee1db68daf77f681959049d/SkiaSharp.Views.Uno.Wasm) - remains 1 other(s).
Loaded dependency (./package_9ce66cf1706e80755ee1db68daf77f681959049d/Uno.UI) - remains 0 other(s).
Debugging hotkey: Shift+Alt+D (when application has focus)
Making persistent: /local
Making persistent: /roaming
Making persistent: /shared
fail: Uno.UI.Dispatching.CoreDispatcher[0]
      Dispatcher unhandled exception
(Line: 25, Col: 22, Idx: 1490) - (Line: 25, Col: 22, Idx: 1490): While scanning a multiline plain scalar, found invalid mapping.
The thread 0x89ec has exited with code 0 (0x0).

The thread 0x6974 has exited with code 0 (0x0).

The thread 0x6870 has exited with code 0 (0x0).

The thread 0x8b18 has exited with code 0 (0x0).`

Please help me understand what might be happening and how to debug this issue.

Thank you!

**Comment by jeromelaban** [comment]
Thanks for the report. Could you post a repro sample of what you're trying to do?

**Comment by DclearEE** [comment, marked as answer]
VS wasn't showing me where the bug was happening but, I was able to set a break point and step through till I hit the bug.
The issue seems to be when I am trying to read a YAML file that I had in my Assets folder.

The problem was fixed when I switched some YAML Deserialization functions to NewtonSoft.JSON functions. And converted the YAML files to JSON files. I was using the package YamlDotNet version 12.0.2 in this project. While it was building for all platforms, only Windows and Android were able to successfully deploy and run.
